src/main/java/net/dries007/tfc/common/blocks/plant/KelpTreeBlock.java

```diff
@@ -0,0 +1,167 @@
+package net.dries007.tfc.common.blocks.plant;
+
+import java.util.Random;
+
+import net.minecraft.block.AbstractBlock;
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockState;
+import net.minecraft.block.SixWayBlock;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.fluid.Fluid;
+import net.minecraft.fluid.FluidState;
+import net.minecraft.fluid.Fluids;
+import net.minecraft.item.BlockItemUseContext;
+import net.minecraft.state.StateContainer;
+import net.minecraft.util.Direction;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.IBlockReader;
+import net.minecraft.world.IWorld;
+import net.minecraft.world.IWorldReader;
+import net.minecraft.world.World;
+import net.minecraft.world.server.ServerWorld;
+
+import net.dries007.tfc.common.TFCTags;
+import net.dries007.tfc.common.fluids.FluidProperty;
+import net.dries007.tfc.common.fluids.IFluidLoggable;
+
+public abstract class KelpTreeBlock extends SixWayBlock implements IFluidLoggable
+{
+    public static KelpTreeBlock create(AbstractBlock.Properties builder, FluidProperty fluid)
+    {
+        return new KelpTreeBlock(builder)
+        {
+            @Override
+            public FluidProperty getFluidPropertyAbstract()
+            {
+                return fluid;
+            }
+        };
+    }
+
+    protected KelpTreeBlock(AbstractBlock.Properties builder)
+    {
+        super(0.3125F, builder);
+        registerDefaultState(stateDefinition.any().setValue(NORTH, Boolean.FALSE).setValue(EAST, Boolean.FALSE).setValue(SOUTH, Boolean.FALSE).setValue(WEST, Boolean.FALSE).setValue(UP, Boolean.FALSE).setValue(DOWN, Boolean.FALSE)
+        .setValue(getFluidProperty(), getFluidProperty().keyFor(Fluids.EMPTY)));
+    }
+
+    @Override
+    public BlockState getStateForPlacement(BlockItemUseContext context)
+    {
+        return this.getStateForPlacement(context.getLevel(), context.getClickedPos());
+    }
+
+    public BlockState getStateForPlacement(IBlockReader world, BlockPos pos)
+    {
+        Block downBlock = world.getBlockState(pos.below()).getBlock();
+        Block upBlock = world.getBlockState(pos.above()).getBlock();
+        Block northBlock = world.getBlockState(pos.north()).getBlock();
+        Block eastBlock = world.getBlockState(pos.east()).getBlock();
+        Block southBlock = world.getBlockState(pos.south()).getBlock();
+        Block westBlock = world.getBlockState(pos.west()).getBlock();
+        return this.defaultBlockState()
+            .setValue(DOWN, downBlock.is(TFCTags.Blocks.KELP_TREE) || downBlock.is(TFCTags.Blocks.SEA_BUSH_PLANTABLE_ON))
+            .setValue(UP, upBlock.is(TFCTags.Blocks.KELP_TREE))
+            .setValue(NORTH, northBlock.is(TFCTags.Blocks.KELP_TREE))
+            .setValue(EAST, eastBlock.is(TFCTags.Blocks.KELP_TREE))
+            .setValue(SOUTH, southBlock.is(TFCTags.Blocks.KELP_TREE))
+            .setValue(WEST, westBlock.is(TFCTags.Blocks.KELP_TREE));
```

Comment:
This could be a `for (Direction direction : Direction.values())` loop and it would be cleaner imo.